### ISSUE TITLE
New Android crash [com.adobe.air.AIRWindowSurfaceView.surfaceChanged]

### ISSUE THREAD

**Issue by megajogos** [airsdk/Adobe-Runtime-Support#823]
Affected Android versions: Android 4, 5, 6 and 7
Brands: 72% Sumsung devices
Adobe Air SDK: AIR 33.1.1.406

Logs:

```
Fatal Exception: java.lang.Error: FATAL EXCEPTION [main]
Unity version     : 2019.4.19f1
Device model      : samsung SM-J510MN
Device fingerprint: samsung/j5xnlteub/j5xnlte:7.1.1/NMF26X/J510MNVJS3BTI1:user/release-keys
```

```
Caused by java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.app.Activity.getSystemService(java.lang.String)' on a null object reference
       at com.adobe.air.AIRWindowSurfaceView.surfaceChanged(AIRWindowSurfaceView.java:768)
       at android.view.SurfaceView.updateWindow(SurfaceView.java:674)
       at android.view.SurfaceView.onWindowVisibilityChanged(SurfaceView.java:258)
       at android.view.View.dispatchWindowVisibilityChanged(View.java:11296)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1293)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1293)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1293)
       at android.view.ViewGroup.dispatchWindowVisibilityChanged(ViewGroup.java:1293)
       at com.android.internal.policy.DecorView.dispatchWindowVisibilityChanged(DecorView.java:461)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1877)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1569)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7301)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:930)
       at android.view.Choreographer.doCallbacks(Choreographer.java:705)
       at android.view.Choreographer.doFrame(Choreographer.java:640)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:916)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6816)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1563)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1451)
```

![image](https://user-images.githubusercontent.com/42586185/116783118-546f5a00-aa63-11eb-8b8b-1ef7a27dd687.png)
![image](https://user-images.githubusercontent.com/42586185/116783130-6650fd00-aa63-11eb-86ed-677ebc21e6ac.png)
![image](https://user-images.githubusercontent.com/42586185/116783140-749f1900-aa63-11eb-9b21-d39caae0ac6f.png)

**Comment by megajogos** [comment]
Hi,

This is our manifest: 
```
<manifest android:installLocation="auto">
    <!-- BEGIN FOR ACHIEVEMENTS/ADMOB/LEADERBORAD -->
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <!-- END FOR ACHIEVEMENTS/ADMOB/LEADERBORAD -->

    <!-- BEGIN PUSH AND FIREBASE PERMISSIONS -->

    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>

   <!-- BADGE -->
    <uses-permission android:name="com.sec.android.provider.badge.permission.READ" />
    <uses-permission android:name="com.sec.android.provider.badge.permission.WRITE" /> 
    <uses-permission android:name="com.htc.launcher.permission.READ_SETTINGS" />
    <uses-permission android:name="com.htc.launcher.permission.UPDATE_SHORTCUT" /> 
    <uses-permission android:name="com.sonyericsson.home.permission.BROADCAST_BADGE" />
    <uses-permission android:name="com.sonymobile.home.permission.PROVIDER_INSERT_BADGE" /> 
    <uses-permission android:name="com.anddoes.launcher.permission.UPDATE_COUNT" /> 
    <uses-permission android:name="com.majeur.launcher.permission.UPDATE_BADGE" /> 
    <uses-permission android:name="com.huawei.android.launcher.permission.CHANGE_BADGE" />
    <uses-permission android:name="com.huawei.android.launcher.permission.READ_SETTINGS" />
    <uses-permission android:name="com.huawei.android.launcher.permission.WRITE_SETTINGS" /> 
    <uses-permission android:name="android.permission.READ_APP_BADGE" /> 
    <uses-permission android:name="com.oppo.launcher.permission.READ_SETTINGS" />
    <uses-permission android:name="com.oppo.launcher.permission.WRITE_SETTINGS" /> 
    <uses-permission android:name="me.everything.badger.permission.BADGE_COUNT_READ" />
    <uses-permission android:name="me.everything.badger.permission.BADGE_COUNT_WRITE" />
	
	<!-- facebook -->
	<uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE" />

    <!-- END PUSH AND FIREBASE SPECIFIC PERMISSIONS -->

   <uses-permission android:name="android.permission.INTERNET"/>
    <uses-permission android:name="android.permission.DISABLE_KEYGUARD"/>
    <uses-permission android:name="android.permission.WAKE_LOCK"/>
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
    <uses-permission android:name="android.permission.NETWORK_STATE"/>
    <uses-permission android:name="android.permission.WIFI_STATE"/>
    <uses-permission android:name="android.permission..BILLING"/>
    <uses-permission android:name="android.permission.VIBRATE"/>
    <uses-permission android:name="android.permission.GET_ACCOUNTS"/>
    <uses-permission android:name="android.permission.GET_TASKS"/>
    <uses-permission android:name="android.permission.READ_PHONE_STATE"/>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <permission android:name="{{android_package}}.permission.C2D_MESSAGE" android:protectionLevel="signature"/>
    <uses-permission android:name="{{android_package}}.permission.C2D_MESSAGE"/>
    <uses-permission android:name="com.google.android.c2dm.permission.RECEIVE"/>
    
    <uses-sdk android:minSdkVersion="14" android:targetSdkVersion="29"/>
    <uses-permission android:name="com.android.vending.BILLING"/>
    <uses-permission android:name="android.permission.CAMERA"/>
    <supports-screens android:smallScreens="true"
                      android:normalScreens="true"
                      android:largeScreens="true"
                      android:xlargeScreens="true"
                      android:requiresSmallestWidthDp="600"/>
    <application android:hardwareAccelerated="true" android:name="androidx.multidex.MultiDexApplication" android:appComponentFactory="androidx.core.app.CoreComponentFactory" android:roundIcon="@mipmap/icon_round">
	
	    <!-- this lets you support devices larger than 16:9 ratio -->

        <meta-data android:name="android.max_aspect" android:value="2.16" />

        <!-- BEGIN ICON FIREBASE SPECS -->
        <meta-data android:name="com.google.firebase.messaging.default_notification_icon"
                   android:resource="@drawable/ic_stat_onesignal_default"/>
        <meta-data android:name="com.google.firebase.messaging.default_notification_color"
                   android:resource="@color/black"/>
        <meta-data android:name="com.google.firebase.messaging.default_notification_channel_id"
                   android:value="default_channel_id"/>
        <!-- END ICON FIREBASE SPECS -->

        <!-- BEGIN DISTRICT GAME SERVICES -->
        <activity
                android:name="com.google.android.gms.auth.api.signin.internal.SignInHubActivity"
                android:excludeFromRecents="true"
                android:exported="false"
                android:windowSoftInputMode="stateAlwaysHidden|adjustPan"
                android:theme="@android:style/Theme.Translucent.NoTitleBar"/>


        <service
                android:name="com.distriqt.extension.jobscheduler.services.TerminateAppJobService"
                android:permission="android.permission.BIND_JOB_SERVICE"
                android:exported="true"/>


        <!-- BEGIN FOR ACHIEVEMENTS/ADMOB/LEADERBORAD   WindowSoftInputMode="stateAlwaysHidden|adjustPan -->
        <service
                android:name="com.google.android.gms.auth.api.signin.RevocationBoundService"
                android:exported="true"
                android:permission="com.google.android.gms.auth.api.signin.permission.REVOCATION_NOTIFICATION"/>
        <!-- END DISTRICT GAME SERVICES -->

        <!-- BEGIN PUSH SPECIFIC -->
        <meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version"/>

		<!-- BEGIN ADMOB -->
		 <meta-data
            android:name="com.google.android.gms.ads.APPLICATION_ID"
            android:value="{{admob_id}}"/>
			
		<activity 
            android:name="com.google.android.gms.ads.AdActivity"
            android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|uiMode|screenSize|smallestScreenSize" 
            android:exported="false"
            android:theme="@android:style/Theme.Translucent" />

        <provider
            android:name="com.google.android.gms.ads.MobileAdsInitProvider"
            android:authorities="{{android_package}}.mobileadsinitprovider"
            android:exported="false"
            android:initOrder="100" />

        <service
            android:name="com.google.android.gms.ads.AdService"
            android:enabled="true"
            android:exported="false" />


        <!-- AndroidX Room já ta em baixo
        <service
            android:name="androidx.room.MultiInstanceInvalidationService"
            android:exported="false" />-->

            
        <!-- AndroidX Work jรก ta la em baixo-->
        <!--<provider
            android:name="androidx.work.impl.WorkManagerInitializer"
            android:authorities="{{android_package}}.workmanager-init"
            android:directBootAware="false"
            android:exported="false"
            android:multiprocess="true" />-->

        <service
            android:name="androidx.work.impl.background.systemalarm.SystemAlarmService"
            android:directBootAware="false"
            android:enabled="@bool/enable_system_alarm_service_default"
            android:exported="false" />
        <service
            android:name="androidx.work.impl.background.systemjob.SystemJobService"
            android:directBootAware="false"
            android:enabled="@bool/enable_system_job_service_default"
            android:exported="true"
            android:permission="android.permission.BIND_JOB_SERVICE" />

        <receiver
            android:name="androidx.work.impl.utils.ForceStopRunnable$BroadcastReceiver"
            android:directBootAware="false"
            android:enabled="true"
            android:exported="false" />
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$BatteryChargingProxy"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.intent.action.ACTION_POWER_CONNECTED" />
                <action android:name="android.intent.action.ACTION_POWER_DISCONNECTED" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$BatteryNotLowProxy"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.intent.action.BATTERY_OKAY" />
                <action android:name="android.intent.action.BATTERY_LOW" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$StorageNotLowProxy"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.intent.action.DEVICE_STORAGE_LOW" />
                <action android:name="android.intent.action.DEVICE_STORAGE_OK" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$NetworkStateProxy"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.net.conn.CONNECTIVITY_CHANGE" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.RescheduleReceiver"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
                <action android:name="android.intent.action.TIME_SET" />
                <action android:name="android.intent.action.TIMEZONE_CHANGED" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxyUpdateReceiver"
            android:directBootAware="false"
            android:enabled="@bool/enable_system_alarm_service_default"
            android:exported="false" >
            <intent-filter>
                <action android:name="androidx.work.impl.background.systemalarm.UpdateProxies" />
            </intent-filter>
        </receiver>
			
			
		<!-- Mediacao facebook -->
		<activity
			android:name="com.facebook.ads.AudienceNetworkActivity"
			android:configChanges="keyboardHidden|orientation|screenSize"
			android:exported="false"
			android:theme="@android:style/Theme.Translucent.NoTitleBar" />

		<provider
			android:name="com.facebook.ads.AudienceNetworkContentProvider"
			android:authorities="{{android_package}}.AudienceNetworkContentProvider"
			android:exported="false" />

		<activity
			android:name="com.facebook.ads.internal.ipc.RemoteANActivity"
			android:configChanges="keyboardHidden|orientation|screenSize"
			android:exported="false"
			android:process=":adnw"
			android:theme="@android:style/Theme.Translucent.NoTitleBar" />

		<service
			android:name="com.facebook.ads.internal.ipc.AdsProcessPriorityService"
			android:exported="false" />
		<service
			android:name="com.facebook.ads.internal.ipc.AdsMessengerService"
			android:exported="false"
			android:process=":adnw" />
		<!-- END Mediacao facebook -->
		<!-- BEGIN Mediacao ironsource -->
		<activity
			android:name="com.ironsource.sdk.controller.ControllerActivity"
			android:configChanges="orientation|screenSize"
			android:hardwareAccelerated="true" />
		<activity
			android:name="com.ironsource.sdk.controller.InterstitialActivity"
			android:configChanges="orientation|screenSize"
			android:hardwareAccelerated="true"
			android:theme="@android:style/Theme.Translucent" />
		<activity
			android:name="com.ironsource.sdk.controller.OpenUrlActivity"
			android:configChanges="orientation|screenSize"
			android:hardwareAccelerated="true"
			android:theme="@android:style/Theme.Translucent" />

		<provider
			android:name="com.ironsource.lifecycle.IronsourceLifecycleProvider"
			android:authorities="{{android_package}}.IronsourceLifecycleProvider" />
		<!-- END Mediacao ironsource -->
		<!-- UNITY ADS -->
		<activity
			android:name="com.unity3d.services.ads.adunit.AdUnitActivity"
			android:configChanges="fontScale|keyboard|keyboardHidden|locale|mnc|mcc|navigation|orientation|screenLayout|screenSize|smallestScreenSize|uiMode|touchscreen"
			android:hardwareAccelerated="true"
			android:theme="@android:style/Theme.NoTitleBar.Fullscreen" />
		<activity
			android:name="com.unity3d.services.ads.adunit.AdUnitTransparentActivity"
			android:configChanges="fontScale|keyboard|keyboardHidden|locale|mnc|mcc|navigation|orientation|screenLayout|screenSize|smallestScreenSize|uiMode|touchscreen"
			android:hardwareAccelerated="true"
			android:theme="@android:style/Theme.Translucent.NoTitleBar.Fullscreen" />
		<activity
			android:name="com.unity3d.services.ads.adunit.AdUnitTransparentSoftwareActivity"
			android:configChanges="fontScale|keyboard|keyboardHidden|locale|mnc|mcc|navigation|orientation|screenLayout|screenSize|smallestScreenSize|uiMode|touchscreen"
			android:hardwareAccelerated="false"
			android:theme="@android:style/Theme.Translucent.NoTitleBar.Fullscreen" />
		<activity
			android:name="com.unity3d.services.ads.adunit.AdUnitSoftwareActivity"
			android:configChanges="fontScale|keyboard|keyboardHidden|locale|mnc|mcc|navigation|orientation|screenLayout|screenSize|smallestScreenSize|uiMode|touchscreen"
			android:hardwareAccelerated="false"
			android:theme="@android:style/Theme.NoTitleBar.Fullscreen" />
		<!-- END ADMOB -->
		
		<!-- com.google.android.play -->
			<activity
				android:name="com.google.android.play.core.missingsplits.PlayCoreMissingSplitsActivity"
				android:enabled="false"
				android:exported="false"
				android:launchMode="singleInstance"
				android:process=":playcore_missing_splits_activity"
				android:stateNotNeeded="true" />
			<activity
				android:name="com.google.android.play.core.common.PlayCoreDialogWrapperActivity"
				android:enabled="false"
				android:exported="false"
				android:stateNotNeeded="true"
				android:theme="@style/Theme.PlayCore.Transparent" />
			<service
				android:name="com.google.android.play.core.assetpacks.AssetPackExtractionService"
				android:enabled="false"
				android:exported="true" />
		<!-- end com.google.android.play -->
		
		<provider
			android:name="com.distriqt.extension.pushnotifications.content.FileProvider"
			android:authorities="{{android_package}}.pushnotificationsfileprovider"
			android:grantUriPermissions="true"
			android:exported="false">
			<meta-data 
				android:name="android.support.FILE_PROVIDER_PATHS" 
				android:resource="@xml/distriqt_pushnotifications_paths" />
		</provider>

        <!-- ONE SIGNAL -->
        <meta-data android:name="com.onesignal.NotificationServiceExtension" android:value="com.distriqt.extension.pushnotifications.onesignal.NotificationServiceExtension" />

		<meta-data
            android:name="com.google.firebase.messaging.default_notification_channel_id"
            android:value="default" />
			
        <receiver
            android:name="com.onesignal.FCMBroadcastReceiver"
            android:permission="com.google.android.c2dm.permission.SEND" >

            <!-- High priority so OneSignal payloads can be filtered from other FCM receivers -->
            <intent-filter android:priority="999" >
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
                <category android:name="{{android_package}}" />
            </intent-filter>
        </receiver>

        <service
            android:name="com.onesignal.HmsMessageServiceOneSignal"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.huawei.push.action.MESSAGING_EVENT" />
            </intent-filter>
        </service>

        <activity
            android:name="com.onesignal.NotificationOpenedActivityHMS"
            android:noHistory="true"
            android:theme="@android:style/Theme.Translucent.NoTitleBar" >
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
            </intent-filter>
        </activity>

        <service android:name="com.onesignal.FCMIntentService" />
        <service
            android:name="com.onesignal.FCMIntentJobService"
            android:permission="android.permission.BIND_JOB_SERVICE" />
        <service
            android:name="com.onesignal.FocusDelaySyncService"
            android:stopWithTask="true" />
        <service
            android:name="com.onesignal.FocusDelaySyncJobService"
            android:permission="android.permission.BIND_JOB_SERVICE" />
        <service
            android:name="com.onesignal.SyncService"
            android:stopWithTask="true" />
        <service
            android:name="com.onesignal.SyncJobService"
            android:permission="android.permission.BIND_JOB_SERVICE" />

        <activity
            android:name="com.onesignal.PermissionsActivity"
            android:theme="@android:style/Theme.Translucent.NoTitleBar" />

        <receiver android:name="com.onesignal.NotificationDismissReceiver" />
        <receiver android:name="com.onesignal.BootUpReceiver" >
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
                <action android:name="android.intent.action.QUICKBOOT_POWERON" />
            </intent-filter>
        </receiver>
        <receiver android:name="com.onesignal.UpgradeReceiver" >
            <intent-filter>
                <action android:name="android.intent.action.{{android_package}}" />
            </intent-filter>
        </receiver>

        <activity
            android:name="com.onesignal.NotificationOpenedReceiver"
            android:noHistory="true"
            android:theme="@android:style/Theme.Translucent.NoTitleBar" />

        <!--
                FirebaseMessagingService performs security checks at runtime,
                but set to not exported to explicitly avoid allowing another app to call it.
        -->
        <service
            android:name="com.google.firebase.messaging.FirebaseMessagingService"
            android:exported="false" >
            <intent-filter android:priority="-500" >
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>

<!-- FIREBASE CORE -->
<meta-data
            android:name="google_analytics_automatic_screen_reporting_enabled"
            android:value="false" />
        
        <activity
            android:name="com.google.android.gms.common.api.GoogleApiActivity"
            android:exported="false"
            android:theme="@android:style/Theme.Translucent.NoTitleBar" />

        <!-- FIREBASE CORE -->
        <receiver
            android:name="com.google.android.gms.measurement.AppMeasurementReceiver"
            android:enabled="true"
            android:exported="false" >
        </receiver>

        <service
            android:name="com.google.android.gms.measurement.AppMeasurementService"
            android:enabled="true"
            android:exported="false" />
        <service
            android:name="com.google.android.gms.measurement.AppMeasurementJobService"
            android:enabled="true"
            android:exported="false"
            android:permission="android.permission.BIND_JOB_SERVICE" />
        <service
            android:name="com.google.firebase.components.ComponentDiscoveryService"
            android:directBootAware="true"
            android:exported="false" >
             <meta-data
				android:name="com.google.firebase.components:com.google.firebase.remoteconfig.RemoteConfigRegistrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />
			<meta-data
				android:name="com.google.firebase.components:com.google.firebase.abt.component.AbtRegistrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />

			<meta-data
				android:name="com.google.firebase.components:com.google.firebase.analytics.connector.internal.AnalyticsConnectorRegistrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />
			<meta-data
				android:name="com.google.firebase.components:com.google.firebase.installations.FirebaseInstallationsRegistrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />
				
			<meta-data
				android:name="com.google.firebase.components:com.google.firebase.crashlytics.CrashlyticsRegistrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />	
                       
			<meta-data
				android:name="com.google.firebase.components:com.google.firebase.iid.Registrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />

			<meta-data
				android:name="com.google.firebase.components:com.google.firebase.messaging.FirebaseMessagingRegistrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />
			<meta-data
				android:name="com.google.firebase.components:com.google.firebase.datatransport.TransportRegistrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />
				
			<meta-data
				android:name="com.google.firebase.components:com.google.firebase.dynamicloading.DynamicLoadingRegistrar"
				android:value="com.google.firebase.components.ComponentRegistrar" />
				
        </service>

        <provider
            android:name="com.google.firebase.provider.FirebaseInitProvider"
            android:authorities="{{android_package}}.firebaseinitprovider"
            android:exported="false"
            android:initOrder="100" />
<!-- FIREBASE CORE -->

<!-- iid -->
        <receiver
            android:name="com.google.firebase.iid.FirebaseInstanceIdReceiver"
            android:exported="true"
            android:permission="com.google.android.c2dm.permission.SEND" >
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
            </intent-filter>
        </receiver>


        <!-- androidx.work -->
        <provider
            android:name="androidx.work.impl.WorkManagerInitializer"
            android:authorities="{{android_package}}.workmanager-init"
            android:directBootAware="false"
            android:exported="false"
            android:multiprocess="true" />

        <service
            android:name="androidx.work.impl.background.systemalarm.SystemAlarmService"
            android:directBootAware="false"
            android:enabled="@bool/enable_system_alarm_service_default"
            android:exported="false" />
        <service
            android:name="androidx.work.impl.background.systemjob.SystemJobService"
            android:directBootAware="false"
            android:enabled="@bool/enable_system_job_service_default"
            android:exported="true"
            android:permission="android.permission.BIND_JOB_SERVICE" />
        <service
            android:name="androidx.work.impl.foreground.SystemForegroundService"
            android:directBootAware="false"
            android:enabled="@bool/enable_system_foreground_service_default"
            android:exported="false" />

        <receiver
            android:name="androidx.work.impl.utils.ForceStopRunnable$BroadcastReceiver"
            android:directBootAware="false"
            android:enabled="true"
            android:exported="false" />
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$BatteryChargingProxy"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.intent.action.ACTION_POWER_CONNECTED" />
                <action android:name="android.intent.action.ACTION_POWER_DISCONNECTED" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$BatteryNotLowProxy"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.intent.action.BATTERY_OKAY" />
                <action android:name="android.intent.action.BATTERY_LOW" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$StorageNotLowProxy"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.intent.action.DEVICE_STORAGE_LOW" />
                <action android:name="android.intent.action.DEVICE_STORAGE_OK" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxy$NetworkStateProxy"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.net.conn.CONNECTIVITY_CHANGE" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.RescheduleReceiver"
            android:directBootAware="false"
            android:enabled="false"
            android:exported="false" >
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
                <action android:name="android.intent.action.TIME_SET" />
                <action android:name="android.intent.action.TIMEZONE_CHANGED" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.background.systemalarm.ConstraintProxyUpdateReceiver"
            android:directBootAware="false"
            android:enabled="@bool/enable_system_alarm_service_default"
            android:exported="false" >
            <intent-filter>
                <action android:name="androidx.work.impl.background.systemalarm.UpdateProxies" />
            </intent-filter>
        </receiver>
        <receiver
            android:name="androidx.work.impl.diagnostics.DiagnosticsReceiver"
            android:directBootAware="false"
            android:enabled="true"
            android:exported="true"
            android:permission="android.permission.DUMP" >
            <intent-filter>
                <action android:name="androidx.work.diagnostics.REQUEST_DIAGNOSTICS" />
            </intent-filter>
        </receiver>

        <!-- com.google.android.datatransport -->
        <!-- duplicado <service
            android:name="com.google.android.datatransport.runtime.backends.TransportBackendDiscovery"
            android:exported="false" >
            <meta-data
                android:name="backend:com.google.android.datatransport.cct.CctBackendFactory"
                android:value="cct" />
        </service>
        <service
            android:name="com.google.android.datatransport.runtime.scheduling.jobscheduling.JobInfoSchedulerService"
            android:exported="false"
            android:permission="android.permission.BIND_JOB_SERVICE" >
        </service>

        <receiver
            android:name="com.google.android.datatransport.runtime.scheduling.jobscheduling.AlarmManagerSchedulerBroadcastReceiver"
            android:exported="false" />
-->
        <service
            android:name="androidx.room.MultiInstanceInvalidationService"
            android:directBootAware="true"
            android:exported="false" />


       

        <!-- END PUSH SPECIFIC -->

        <meta-data android:name="{{android_package}}" android:value="{{facebook_app_id}}"/>
        <meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version"/>
        <meta-data android:name="com.google.android.gms.games.APP_ID" android:value="\ {{google_games_app_id}}"/>

        <activity android:launchMode="standard">
			<intent-filter>
				<action android:name="android.intent.action.MAIN" />
				<category android:name="android.intent.category.LAUNCHER" />
			</intent-filter>
			<intent-filter android:autoVerify="true">
				<action android:name="android.intent.action.VIEW" />
				<category android:name="android.intent.category.BROWSABLE" />
				<category android:name="android.intent.category.DEFAULT" />
				<data android:scheme="megajogos" android:host="{{tag}}" />
				<data android:scheme="https" android:host="www.megajogos.com.br" android:pathPrefix="/deeplink-{{tag}}" />
				<data android:scheme="https" android:host="www.megajogos.com.br" android:pathPrefix="/deeplink" />
			</intent-filter>
		</activity>

        <!-- BEGIN FACEBOOK -->
		
		<!-- IMPORTANT: Do not remove the backslash and space at the beginning of the following value string. -->
        <meta-data android:name="com.facebook.sdk.ApplicationId" android:value="\ {{facebook_app_id}}"/>

        <!-- OPTIONAL SETTINGS -->
        <meta-data android:name="com.facebook.sdk.AutoLogAppEventsEnabled" android:value="true"/>
        <meta-data android:name="com.facebook.sdk.AutoInitEnabled" android:value="true"/>
        <meta-data android:name="com.facebook.sdk.AdvertiserIDCollectionEnabled" android:value="true"/>

        <activity
            android:name="com.facebook.FacebookActivity"
            android:configChanges="keyboard|keyboardHidden|screenLayout|screenSize|orientation"
            android:theme="@android:style/Theme.Translucent.NoTitleBar" />
        <activity android:name="com.facebook.CustomTabMainActivity" />
        <activity
            android:name="com.facebook.CustomTabActivity"
            android:exported="true" >
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
                <data android:scheme="fb{{facebook_app_id}}" />

            </intent-filter>
        </activity>

        <provider
            android:name="com.facebook.internal.FacebookInitProvider"
            android:authorities="{{android_package}}.FacebookInitProvider"
            android:exported="false" />

        <receiver
            android:name="com.facebook.CurrentAccessTokenExpirationBroadcastReceiver"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.facebook.sdk.ACTION_CURRENT_ACCESS_TOKEN_CHANGED" />
            </intent-filter>
        </receiver>
		
		<!-- Share -->
		<provider android:authorities="com.facebook.app.FacebookContentProvider{{facebook_app_id}}"
			android:name="com.facebook.FacebookContentProvider"
			android:exported="true"/>
			
        <!-- END FACEBOOK -->

        <activity android:label="Login" android:name="com.facebook.LoginActivity"
                  android:theme="@android:style/Theme.Translucent.NoTitleBar"/>

        <!-- BEGIN crashlytics -->
        <!-- Firebase Crashlytics -->
		<meta-data android:name="firebase_crashlytics_collection_enabled" android:value="false" />
		<activity android:name="com.distriqt.extension.firebase.crashlytics.activities.CrashActivity" />

		<!-- datatransport -->
		<service
			android:name="com.google.android.datatransport.runtime.backends.TransportBackendDiscovery"
			android:exported="false" >
			<meta-data
				android:name="backend:com.google.android.datatransport.cct.CctBackendFactory"
				android:value="cct" />
		</service>
		<service
			android:name="com.google.android.datatransport.runtime.scheduling.jobscheduling.JobInfoSchedulerService"
			android:exported="false"
			android:permission="android.permission.BIND_JOB_SERVICE" >
		</service>
		<receiver
			android:name="com.google.android.datatransport.runtime.scheduling.jobscheduling.AlarmManagerSchedulerBroadcastReceiver"
			android:exported="false" />
        <!-- END crashlytics -->

        <!-- BEGIN BILLING -->
        <activity
                android:name="com.distriqt.extension.inappbilling.activities.ProductViewActivity"
                android:theme="@android:style/Theme.Translucent.NoTitleBar"/>

        <activity
                android:name="com.android.billingclient.api.ProxyBillingActivity"
                android:configChanges="keyboard|keyboardHidden|screenLayout|screenSize|orientation"
                android:theme="@android:style/Theme.Translucent.NoTitleBar"/>
        <!-- END BILLING -->

    </application>
</manifest>
```

We always replace some variables like: {{android_package}}, {{admob_id}} and {{facebook_app_id}}

**Comment by ajwfrost** [comment]
Thanks for the report - we've got a fix in for this for the next release..
cheers

**Comment by megajogos** [comment]
When will be the next release?

**Comment by ajwfrost** [comment]
We're looking at the end of this month for the next release, to include all the updates for Android that are required for the API level 30.

**Comment by ajwfrost** [comment]
Should be fixed in 33.1.1.533, please check and re-open if necessary.
Thanks
